### DISCUSSION TITLE
Lesson 7: How does brownie gets the ABI contract?

### DISCUSSION THREAD

**Discussion by andrescoppo** [smartcontractkit/full-blockchain-solidity-course-py#1311] [Q&A]
This maybe is a theoretical question, but I would appreciate if somebody can answer me.

When we are making the helpful_scripts for lesson 7 we write:

```Python

contract_to_mock = {"eth_usd_price_feed": MockV3Aggregator}

def get_contract(contract_name):
 """This function will grab the contract addresses from the brownie config if defined, otherwise, it will deploy a mock version of that contract, and return that mock contract.

    Args:
        contract_name (string)

    Returns:
        brownie.network.contract.ProjectContract: The most recently deployed version of this contract.
"""

    contract_type = contract_to_mock[contract_name]
    if network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
        if len(contract_type) <= 0:
            deploy_mocks()
        contract = contract_type[-1].address

    else:
        contract_address = config["networks"][network.show_active()][contract_name]
        contract=Contract.from_abi(contract_type._name, contract_address, `contract_type.abi)`
```

From where brownie gets the contract_type.abi input? 

Thank you in advance.

**Comment by cromewar** [comment]
Hello @andrescoppo 
It depends if the contract is already deployed into a blockchain brownie gets the abi directly from it based on the contract address.
If you are using a local development environment the abi of the contract is on the build folder amongst the other contracts on your project.

**Reply by andrescoppo** [reply]
but in the case of my  last line code: There, my contract has not been deployed yet and it will be deployed on a testnet. From where brownie gets or generate the ABI?

**Reply by cromewar** [reply]
When your run `brownie compile` the ABI lives on the build folder as a JSON file.
